src/collection-view.js

```diff
@@ -125,11 +125,75 @@ const CollectionView = Backbone.View.extend({
     }
   },
 
-  // get the child view by model it holds, and remove it
-  _onCollectionRemove(model) {
-    const view = this.children.findByModel(model);
-    this.removeChildView(view);
-    this._checkEmpty();
+  // Handle collection update model removals
+  _onCollectionUpdate(collection, {changes: {removed = []} = {}} = {}) {
```

Comment:
I dunno if the defaults are good in this instance because of the transpilation.

Could be
```js
_onCollectionUpdate(collection, options) {
  if(_.isArray(options.removed)) { // you already check the length in _removeChildViews
```

instead it's 
```js
_onCollectionUpdate(collection) {
    var _ref = arguments.length <= 1 || arguments[1] === undefined ? {} : arguments[1];

    var _ref$changes = _ref.changes;
    _ref$changes = _ref$changes === undefined ? {} : _ref$changes;
    var _ref$changes$removed = _ref$changes.removed;
    var removed = _ref$changes$removed === undefined ? [] : _ref$changes$removed;
if (_.isArray(removed) {
```